core/eval.go

```diff
@@ -89,8 +93,27 @@ func evalSET(args []string) []byte {
 			if err != nil {
 				return Encode(errors.New("ERR value is not an integer or out of range"), false)
 			}
+			if exDurationSec < 0 {
+				return Encode(errors.New("ERR invalid expire time in 'set' command"), false)
+			}
 			exDurationMs = exDurationSec * 1000
+		case "PX", "px":
```

Comment:
The PX and EX options are same except for the conversion part in EX and the code is part of the same file. Should we combine them?

Reply:
You mean having a single case for both, and checking for "px" inside that case?